data-structures/stacksandqueues/src/main/java/stacksandqueues/Queue.java

```diff
@@ -0,0 +1,92 @@
+package stacksandqueues;
+
+public class Queue {
+/************
+ * Instance Variables
+ * */
+    private Node front;
+    private Node rear;
+    private int size;
+
+
+/************
+ * Constructors
+ * */
+    public Queue() {
+        this.front = null;
+        this.rear = null;
+        this.size = 0;
+    }
+
+
+/************
+ * Getters/Setters
+ * */
+    public Node getFront() { return this.front; }
+    public Node getRear() { return this.rear; }
+    public int getSize() { return this.size; }
+
+    public void setFront(Node f) { this.front = f; }
+    public void setRear(Node r) { this.rear = r; }
+    public void setSize(int s) { this.size = (s == 1) ? this.size + 1 : this.size - 1; }
+
+
+/************
+ * Additional Instance Methods
+ * */
+    // The enqueue method takes any value as an argument and adds a new node with that value to the back of the queue
+    // with an O(1) Time performance
+    public void enqueue(Object value) {
+        try {
+            Node node = new Node(value);
+
+            if (getFront() == null) {
+                setFront(node);
+                setRear(node);
+            } else {
+                getRear().setNext(node);
+                setRear(node);
+            }
+
+            setSize(1);
```

Comment:
The size of your queue will never be more than `1`! No matter how many times you enqueue this line will always set the size to `1`